src/main/java/com/kodilla/ecommercee/domain/UserDto.java

```diff
@@ -3,17 +3,41 @@
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.ArrayList;
 import java.util.Date;
+import java.util.List;
 
 @Getter
 @AllArgsConstructor
 public class UserDto {
```

Comment:
a czy ta klasa nie powinna bardziej JSON-a przypominać którego dostaniemy z frontendu? nie będzie wtedy danych do expirationTime, listOfOrders

Reply:
Myślałem o tym ale jeżeli Dto obetniemy z tych informacji to nie bedzie mozna ich wyciagac z bazy przez co nie bedziemy mieć dostępu do tych informacji. Można oczywiśćie zrobić osobne zapytanie tylko  o czas i o listę zamówień. Założyłem że te informacje we stepnej fazie projektu są minimum które musimy mieć a my rozszerzyliśmy to o dodatkowe informacje. Jeżeli to są ograniczenia i JSON nie powinien dawać więcej informacji niż to jest w założeniach to faktycznie trzeba będzie to ograniczyć.

Reply:
> a czy ta klasa nie powinna bardziej JSON-a przypominać którego dostaniemy z frontendu? nie będzie wtedy danych do expirationTime, listOfOrders

Myślałem o tym ale jeżeli Dto obetniemy z tych informacji to nie bedzie mozna ich wyciagac z bazy przez co nie bedziemy mieć dostępu do tych informacji. Można oczywiśćie zrobić osobne zapytanie tylko o czas i o listę zamówień. Założyłem że te informacje we stepnej fazie projektu są minimum które musimy mieć a my rozszerzyliśmy to o dodatkowe informacje. Jeżeli to są ograniczenia i JSON nie powinien dawać więcej informacji niż to jest w założeniach to faktycznie trzeba będzie to ograniczyć.

Reply:
czemu nie będzie ich można wyciągnąć z bazy? expirationTime będzie można ustawić np w ciele kostruktora User (bez podawania wartości w argumencie konstruktora) i podczas tworzenia nowego użytkownika, będzie miał on już ten expirationTime utworzony na pierwszą sesję zakupową. Potem każdorazowo podczas logowania ta wartość będzie ustawiana na nowo. Wszystko można wyciągnąć getterami, a sama listOfOrders będzie się aktualizować poprzez mapowanie. Tak mi się wydaje oczywiście :)